ckanext/googleanalytics/views.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+
+import hashlib
+import logging
+import six
+
+from flask import Blueprint
+from werkzeug.utils import import_string
+
+import ckan.logic as logic
+import ckan.plugins.toolkit as tk
+import ckan.views.api as api
+import ckan.views.resource as resource
+
+from ckan.common import g
+
+CONFIG_HANDLER_PATH = "googleanalytics.download_handler"
+
+log = logging.getLogger(__name__)
+ga = Blueprint("google_analytics", "google_analytics")
+
+
+def action(logic_function, ver=api.API_MAX_VERSION):
+    try:
+        function = logic.get_action(logic_function)
+        side_effect_free = getattr(function, "side_effect_free", False)
+        request_data = api._get_request_data(try_url_params=side_effect_free)
+        if isinstance(request_data, dict):
+            id = request_data.get("id", "")
+            if "q" in request_data:
+                id = request_data["q"]
+            if "query" in request_data:
+                id = request_data[u"query"]
+            _post_analytics(g.user, "CKAN API Request", logic_function, "", id)
+    except Exception as e:
+        log.debug(e)
+        pass
+
+    return api.action(logic_function, ver)
+
+
+ga.add_url_rule(
+    "/api/action/<logic_function>",
+    methods=["GET", "POST"],
+    view_func=action,
+)
+ga.add_url_rule(
+    u"/<int(min=3, max={0}):ver>/action/<logic_function>".format(
+        api.API_MAX_VERSION
+    ),
+    methods=["GET", "POST"],
+    view_func=action,
+)
+
+
+def download(id, resource_id, filename=None, package_type="dataset"):
```

Comment:
This function should also post multiple analytics based on resource id, package name, and organization name.

https://github.com/OpenGov-OpenData/ckanext-googleanalytics/blob/97d142a202149817c2c3d91eb5418903255f4d8f/ckanext/googleanalytics/plugin.py#L64-L113